tests/categorical/test_contingency.py

```diff
@@ -0,0 +1,198 @@
+"""
+Test functions for contingency tables
+"""
+
+import numpy as np
+import pytest
+import xarray as xr
+
+import scores
+
+try:
+    import dask
+    import dask.array
+except:  # noqa: E722 allow bare except here # pylint: disable=bare-except  # pragma: no cover
+    dask = "Unavailable"  # type: ignore  # pylint: disable=invalid-name  # pragma: no cover
+
+# Provides a basic forecast data structure in three dimensions
+simple_forecast = xr.DataArray(
+    [
+        [
+            [0.9, 0.0, 5],
+            [0.7, 1.4, 2.8],
+            [0.4, 0.5, 2.3],
+        ],
+        [
+            [1.9, 1.0, 1.5],
+            [1.7, 2.4, 1.1],
+            [1.4, 1.5, 3.3],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+# Within 0.1 or 0.2 of the forecast in all cases except one
+# Can be used to find some exact matches, and some close matches
+simple_obs = xr.DataArray(
+    [
+        [
+            [0.9, 0.0, 5],
+            [0.7, 1.3, 2.7],
+            [0.3, 0.4, 2.2],
+        ],
+        [
+            [1.7, 1.2, 1.7],
+            [1.7, 2.2, 3.9],
+            [1.6, 1.2, 9.9],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+# This truth table shows where the forecast and obs match precisely
+exact_matches = xr.DataArray(
+    [
+        [
+            [True, True, True],
+            [True, False, False],
+            [False, False, False],
+        ],
+        [
+            [False, False, False],
+            [True, False, False],
+            [False, False, False],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+# This truth table shows where the forecast and obs match within 0.2
+somewhat_near_matches = xr.DataArray(
+    [
+        [
+            [True, True, True],
+            [True, True, True],
+            [True, True, True],
+        ],
+        [
+            [True, True, True],
+            [True, True, False],
+            [True, False, False],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+
+def test_str_view():
+    """
+    Smoke test for string representation
+    """
+    match = scores.categorical.ThresholdEventOperator()
+    table = match.make_table(simple_forecast, simple_obs, event_threshold=1.3)
+    _stringrep = str(table)
+
+
+def test_categorical_table():
+    """
+    Test the basic calculations of the contingency table
+    """
+    match = scores.categorical.ThresholdEventOperator(default_event_threshold=1.3)
+    table = match.make_table(simple_forecast, simple_obs)
```

Comment:
Can you add a test where you use a different `op_fn`?

Reply:
I can, but what should I be testing for exactly? Is the purpose to test the score calculations in more numerical situations, or to smoke test or provide more coverage of the event operator class itself?

Reply:
Thanks, done.